api/api/validators/evaluate_team_set_validator.py

```diff
@@ -0,0 +1,199 @@
+from typing import Dict, Any
+
+from schema import Schema, Optional, Or, SchemaError
+
+from api.api.validators.interface import Validator
+from api.models.enums import RequirementOperator, Relationship
+from api.api.utils.relationship import get_relationship_str
+
+
+class EvaluateTeamSetValidator(Validator):
+    def validate(self):
+        self._validate_schema()
+        self.validate_team_set()
+        self.validate_metrics()
+
+    def _validate_schema(self):
+        Schema(
+            {
+                "team_set": dict,
+                "metrics": dict,
+            }
+        ).validate(self.data)
+
+    def validate_team_set(self):
+        team_set = self.data.get("team_set", {})
+
+        if "id" in team_set:
+            Schema(str).validate(team_set["id"])
+            # Check if team_set[id] is a number
+            try:
+                Schema(int).validate(int(team_set["id"]))
+            except ValueError:
+                raise ValueError("team_set[id] must be a number")
+
+            Schema(str).validate(team_set["name"])
```

Comment:
nit;

you don't actually need a name to calculate the metrics though - we should think about the input schema here more thoroughly at some point. As long as we keep the endpoint's name consistent we can change the schema freely imo